compiler/scriptconfig.nim

```diff
@@ -159,8 +159,11 @@ proc runNimScript*(cache: IdentCache; scriptName: string;
 
   defineSymbol(conf.symbols, "nimscript")
   defineSymbol(conf.symbols, "nimconfig")
-  registerPass(graph, semPass)
-  registerPass(graph, evalPass)
+  var registeredPasses {.global.} = false
```

Comment:
I think better is to check ``graph`` whether it already has these passes.

Reply:
I would love to, but `registerPass` [operates on a global var][0], not `graph` (no idea why). This would not be an issue if it operated on `graph`, since it's created everytime the proc run.

[0]: https://github.com/nim-lang/Nim/blob/1074cc1fb9bd0a69f1990f936dd82313cfb2c2c8/compiler/passes.nim#L67

Reply:
well then that's the real bug. We need to attach it to the ModuleGraph properly.

Reply:
fixed in https://github.com/nim-lang/Nim/pull/9259 and https://github.com/nim-lang/Nim/pull/9265